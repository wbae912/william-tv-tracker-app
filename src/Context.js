import React from 'react';

const tvContext = React.createContext({
  shows: [],
  deleteTvShow: () => {},
  addTvShow: () => {}
})

export default tvContext;