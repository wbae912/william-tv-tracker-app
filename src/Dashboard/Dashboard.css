.dashboard-section {
  text-align: center;
  color: white;
  font-family: 'Montserrat', sans-serif;
  padding-top: 65px;
  margin-top: 20px;
  margin-bottom: 70px;
}

.dashboard-title {
  margin-bottom: 40px;
  font-size: 34px;
}

.total-shows {
  font-size: 24px;
  font-weight: bold;
  color: white;
  width: 90%;
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 10px;
}

#underline-total {
  display: none;
}

.chart-title {
  margin-bottom: 2%;
  font-size: 21px;
}

.underline-mini {
  border: 0;
  height: 2px;
  width: 3rem;
  border-top: 2px solid white;
  margin-top: 0;
  margin-bottom: 0;
}

.chart-div {
  margin-top: 10px;
}

.chart {
  display: flex;
  justify-content: center;
  margin-bottom: 40px;
  margin-top: 10px;
}

.rating-chart {
  margin-bottom: 40px;
  margin-top: 10px;
}

@media only screen and (min-width: 750px) {
  .dashboard-title {
    margin-bottom: 50px;
    font-size: 36px;
  }
  .total-shows {
    font-size: 28px;
    margin-bottom: 50px;
  }
  .chart-title {
    font-size: 23px;
  }
  .chart-div {
    margin-bottom: 40px;
  }
  .underline-mini {
    margin-bottom: 10px;
  }
}

@media only screen and (min-width: 1300px) {
  .dashboard-title {
    margin-bottom: 50px;
    font-size: 40px;
  }
  .total-shows {
    font-size: 30px;
  }
  .chart-title {
    font-size: 25px;
  }
  #chart-one, #chart-two {
    width: 40%;
    display: inline-block;
  }
  .underline-mini {
    margin-bottom: 20px;
  }
  .chart-flex {
    display: flex;
    justify-content: space-evenly;
  }
}