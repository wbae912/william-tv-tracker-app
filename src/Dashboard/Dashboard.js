import React, { Component } from 'react';
import tvContext from '../Context';
import RatingChart from '../RatingChart/RatingChart';
import GenreChart from '../GenreChart/GenreChart';
import StatusChart from '../StatusChart/StatusChart';
import Chart from 'chart.js';
import './Dashboard.css'

//Global styling for charts
Chart.defaults.global.legend.display = false;

export default class Dashboard extends Component {
  static contextType = tvContext;

  constructor(props) {
    super(props)
  
    this.state = {
       error: null
    }
  }
  
  
  componentDidMount() {
    this.context.getAllShows()
    .catch(res => {
      this.setState({
        error: res.error
      })
    })
  }

  render() {
    const totalShows = this.context.shows.length; 

    return (
      <section className="dashboard-section"> 
        <div role="alert" className="error-bigger">
          {this.state.error && <p className='red-bigger'>{this.state.error}</p>}
        </div>
        <h1 className="dashboard-title">My Dashboard</h1>
        <h2 className="total-shows">Total TV Shows: {totalShows}</h2>
        <hr className="underline-mini" id="underline-total" />
        <div className="chart-flex">
          <div className="chart-div" id="chart-one">
            <h3 className="chart-title">TV Shows per Status</h3>
            <hr className="underline-mini" />
            <div className="chart">
              <StatusChart />
            </div>
          </div>
          <div className="chart-div" id="chart-two">
            <h3 className="chart-title">TV Shows per Rating</h3>
            <hr className="underline-mini" />
            <div className="chart">
              <RatingChart />
            </div>
          </div>
        </div>
        <div className="chart-div" id="chart-three">
          <h3 className="chart-title">TV Shows per Genre</h3>
          <hr className="underline-mini" />
          <div className="chart">
            <GenreChart />
          </div>
        </div>
      </section>
    )
  }
}

//reset the error in more places