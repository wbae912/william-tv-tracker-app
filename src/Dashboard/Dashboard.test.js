import React from 'react';
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import Dashboard from './Dashboard';
import Context from '../Context';
import TokenService from '../services/token-service';
import renderer from 'react-test-renderer';

describe(`Dashboard component`, () => {
  const shows = [
      {
        "id": "1",
        "tv_title": "Example",
        "status": "Completed",
        "season_number": "1",
        "episode_number": "1",
        "rating": "3",
        "genre": "Action",
        "description": "Lorem ipsum",
        "review": "Lorem ipsum"
      },
      {
        "id": "2",
        "tv_title": "Example 2",
        "status": "Completed",
        "season_number": "1",
        "episode_number": "1",
        "rating": "3",
        "genre": "Action",
        "description": "Lorem ipsum",
        "review": "Lorem ipsum"
      }
    ];
  
  const getAllShows = () => {
    return fetch(`http://localhost:8000/api/shows/all`, {
      headers: {
      'authorization': `bearer ${TokenService.getAuthToken()}`
      }
    })
    .then(res => {
      if(!res.ok) {
        return res.json().then(err => Promise.reject(err));
      }
      return res.json();
    })
    .then(data => {
      this.setState({
        shows: data,
        isLoggedIn: true
      })
    })
  }
  
  it('renders component without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <Context.Provider value={{getAllShows, shows}}><Dashboard /></Context.Provider>
      </MemoryRouter>
      , div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders a .Dashboard by default', () => {
    const wrapper = shallow(<Context.Provider value={{getAllShows, shows}}><Dashboard /></Context.Provider>)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
  // it('renders UI as expected', () => {
  //   const tree = renderer
  //     .create(<Context.Provider value={{getAllShows, shows}}><Dashboard /></Context.Provider>)
  //     .toJSON();
  //     expect(tree).toMatchSnapshot();
  // })
});