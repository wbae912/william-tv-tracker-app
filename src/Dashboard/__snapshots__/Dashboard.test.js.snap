// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Completed component renders a .Dashboard by default 1`] = `<Dashboard />`;
